segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/DebugStreamSegmentContainer.java

```diff
@@ -0,0 +1,74 @@
+import io.pravega.segmentstore.server.WriterFactory;
+import io.pravega.segmentstore.server.attributes.AttributeIndexFactory;
+import io.pravega.segmentstore.storage.StorageFactory;
+import io.pravega.segmentstore.server.SegmentContainerExtension;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+@Slf4j
+public class DebugStreamSegmentContainer extends StreamSegmentContainer implements DebugSegmentContainer {
+    private static final Duration TIMEOUT = Duration.ofMinutes(1);
+    private final ContainerConfig config;
+
+    /**
+     * Creates a new instance of the StreamSegmentContainer class.
+     *
+     * @param streamSegmentContainerId The Id of the StreamSegmentContainer.
+     * @param config                   The ContainerConfig to use for this StreamSegmentContainer.
+     * @param durableLogFactory        The DurableLogFactory to use to create DurableLogs.
+     * @param readIndexFactory         The ReadIndexFactory to use to create Read Indices.
+     * @param attributeIndexFactory    The AttributeIndexFactory to use to create Attribute Indices.
+     * @param writerFactory            The WriterFactory to use to create Writers.
+     * @param storageFactory           The StorageFactory to use to create Storage Adapters.
+     * @param createExtensions         A Function that, given an instance of this class, will create the set of
+     *                                 {@link SegmentContainerExtension}s to be associated with that instance.
+     * @param executor                 An Executor that can be used to run async tasks.
+     */
+    DebugStreamSegmentContainer(int streamSegmentContainerId, ContainerConfig config, OperationLogFactory durableLogFactory,
+                                ReadIndexFactory readIndexFactory, AttributeIndexFactory attributeIndexFactory,
+                                WriterFactory writerFactory, StorageFactory storageFactory,
+                                SegmentContainerFactory.CreateExtensions createExtensions, ScheduledExecutorService executor) {
+        super(streamSegmentContainerId, config, durableLogFactory, readIndexFactory, attributeIndexFactory, writerFactory,
+                storageFactory, createExtensions, executor);
+        this.config = config;
+    }
+
+    /**
+     * Creates a segment with given properties.
+     * @param streamSegmentName         Name of the segment to be created.
+     * @param length                    Length of the segment to be created.
+     * @param isSealed                  Sealed status of the segment to be created.
+     * @return                          A newly created segment.
+     */
+    @Override
+    public CompletableFuture<Void> createStreamSegment(String streamSegmentName, long length, boolean isSealed) {
+        StreamSegmentInformation segmentProp = StreamSegmentInformation.builder()
```

Comment:
Create a new method in `MetadataStore.SegmentInfo` and name it `recoveredSegment` where you hide away all these details. Then revert back the visibility of both `MetadataStore.SegmentInfo` and `MetadataStore.SegmentInfo.serialize` (They are in the same package as this so you don't need the extra `public` visibility).

Reply:
Done.